docker/src/main/java/clocker/docker/location/DockerContainerLocation.java

```diff
@@ -269,13 +273,100 @@ private void parseDockerCallback(String commandString) {
     }
 
     @Override
+    public int copyTo(final Map<String,?> props, final InputStream src, final long filesize, final String destination) {
+        Map<String,?> nonPortProps = Maps.filterKeys(props, Predicates.not(Predicates.containsPattern("port")));
+        boolean entitySsh = Boolean.TRUE.equals(entity.config().get(DockerContainer.DOCKER_USE_SSH));
+        boolean dockerSsh = Boolean.TRUE.equals(getOwner().config().get(DockerContainer.DOCKER_USE_SSH));
+        if (entitySsh && dockerSsh) {
+            return super.copyTo(nonPortProps, src, filesize, destination);
+        } else {
+            return copyTo(props, src, destination);
+        }
+    }
+
+    @Override
+    public int copyTo(final Map<String,?> props, final InputStream src, final String destination) {
+        Map<String,?> nonPortProps = Maps.filterKeys(props, Predicates.not(Predicates.containsPattern("port")));
+        boolean entitySsh = Boolean.TRUE.equals(entity.config().get(DockerContainer.DOCKER_USE_SSH));
+        boolean dockerSsh = Boolean.TRUE.equals(getOwner().config().get(DockerContainer.DOCKER_USE_SSH));
+        if (entitySsh && dockerSsh) {
+            return super.copyTo(nonPortProps, src, destination);
+        } else {
+            File tmp = null;
+            try {
+                tmp = File.createTempFile(dockerContainer.getId(), Urls.getBasename(destination));
+                hostMachine.copyTo(nonPortProps, src, tmp.getAbsolutePath());
```

Comment:
This seems wrong: we create a temp file on the *Clocker* server, and then we use that file's path as the target on the *docker host*. I can see that if the Clocker and docker host are similarly configured, then this will work. But if for example someone tried to run Clocker on windows (or on OS X with a different default tmp directory) then the path would not be applicable on the docker host. Why don't we just construct a name like `org.apache.brooklyn.util.os.Os.mergePathsUnix("/tmp", dockerContainer.getId() + Urls.getBasename(destination))` (possibly with some random id as well)?

Same for the other `copyTo` method below.